src/OpenLoco/src/GameCommands/GameCommands.h

```diff
@@ -696,28 +696,56 @@ namespace OpenLoco::GameCommands
         }
     };
 
-    // Change company colour scheme
-    inline void do_19(int8_t isPrimary, int8_t value, int8_t colourType, int8_t setColourMode, CompanyId companyId)
-    {
-        registers regs;
-        regs.bl = Flags::apply;
-        regs.cl = colourType;           // vehicle type or main
-        regs.dh = setColourMode;        // [ 0, 1 ] -- 0 = set colour, 1 = toggle enabled/disabled;
-        regs.dl = enumValue(companyId); // company id
+    struct ChangeCompanyColourSchemeArgs
+    {
+        static constexpr auto command = GameCommand::changeCompanyColourScheme;
+
+        ChangeCompanyColourSchemeArgs() = default;
+        explicit ChangeCompanyColourSchemeArgs(const registers& regs)
+            : companyId(CompanyId(regs.dl))
+            , isPrimary()
+            , value(regs.al)
+            , colourType(regs.cl)
+            , setColourMode(regs.dh)
+        {
+            if (setColourMode == 0)
+            {
+                isPrimary = regs.ah == 0;
+            }
+            else if (setColourMode == 1)
+            {
+                isPrimary = regs.al;
```

Comment:
isPrimary is only used for setColourMode 0. Remove this line.

Reply:
Will do